src/test/logic/token/utils/tokenHelpers.test.ts

```diff
@@ -0,0 +1,193 @@
+      name: "Dai",
+      symbol,
+      decimals,
+      logoUri: "https://gnosis-safe-token-logos.s3.amazonaws.com/0x5592EC0cfb4dbc12D3aB100b257153436a1f0FEa.png",
+      balance: 0
+    })
+    const expectedResult = {
+      decimals,
+      symbol
+    }
+
+
+    const fetchTokens = require('src/logic/tokens/store/actions/fetchTokens')
+    const spy = fetchTokens.getTokenInfos.mockImplementationOnce(() => token)
+
+    // when
+    const result = await getERC20DecimalsAndSymbol(tokenAddress)
+
+    // then
+    expect(result).toEqual(expectedResult)
+    expect(spy).toHaveBeenCalled()
+  })
+  it('It should return default value decimals: 18, symbol: UNKNOWN if given a token address and if there is an error fetching the data', async () => {
+    // given
+    const tokenAddress = "0x5592EC0cfb4dbc12D3aB100b257153436a1f0FEa"
+    const decimals = Number(18)
+    const symbol = "UNKNOWN"
+
+    const expectedResult = {
+      decimals,
+      symbol
+    }
+
+
+    const fetchTokens = require('src/logic/tokens/store/actions/fetchTokens')
+    const spy = fetchTokens.getTokenInfos.mockImplementationOnce(() => {throw new Error()})
+    console.error = jest.fn();
+    const spyConsole = jest.spyOn(console, 'error').mockImplementation()
+
+    // when
+    const result = await getERC20DecimalsAndSymbol(tokenAddress)
+
+    // then
+    expect(result).toEqual(expectedResult)
+    expect(spy).toHaveBeenCalled()
+    expect(spyConsole).toHaveBeenCalled()
+  })
+  it('It should fetch token information from the blockchain if given a token address and if the information is not stored', async () => {
+    // given
+    const tokenAddress = "0x5592EC0cfb4dbc12D3aB100b257153436a1f0FEa"
+    const decimals = Number(18)
+    const symbol = "DAI"
+    const expectedResult = {
+      decimals,
+      symbol
+    }
+
+
+    const fetchTokens = require('src/logic/tokens/store/actions/fetchTokens')
+    const generateBatchRequests = require('src/logic/contracts/generateBatchRequests')
+    const spyTokenInfos = fetchTokens.getTokenInfos.mockImplementationOnce(() => null)
+
+    const spyGenerateBatchRequest = generateBatchRequests.default.mockImplementationOnce(() => [decimals, symbol])
+
+    // when
+    const result = await getERC20DecimalsAndSymbol(tokenAddress)
+
+    // then
+    expect(result).toEqual(expectedResult)
+    expect(spyTokenInfos).toHaveBeenCalled()
+    expect(spyGenerateBatchRequest).toHaveBeenCalled()
+  })
+})
+
+describe('isERC721Contract', () => {
+  afterAll(() => {
+    jest.unmock('src/logic/tokens/store/actions/fetchTokens')
+  })
+  beforeEach(() => {
+    jest.mock('src/logic/tokens/store/actions/fetchTokens')
+  })
+  it('It should return false if given an random non-erc721 contract address', async () => {
+    // given
+    const contractAddress = "0x5592EC0cfb4dbc12D3aB100b257153436a1f0FEa" // DAI Address
+    const expectedResult = false
+
+    // when
+    const result = await isERC721Contract(contractAddress)
```

Comment:
how does it work? This function needs to call a method on a contract to work properly but I couldn't find any mock.